### DISCUSSION TITLE
Question sur One Shot Trigger For Reed Switch

### DISCUSSION THREAD

**Discussion by fguiet** [PricelessToolkit/MailBoxGuard#33] [General]
Salut Priceless,

Dans le schéma Electronic et la partie One Shot Trigger For Reed Switch, je ne comprends pas pourquoi tu as besoin d'une diode, des résistances R10, R8 et du condensateur C1...

En enlevant les composants dont ci-dessous, Si on le voltage diminue lorsque la reedswitch s'ouvre on aura un LOW au niveau du mostfet et le courant pourra passer...donc ca devrait fonctionner?

![image](https://github.com/PricelessToolkit/MailBoxGuard/assets/13849639/50ac64b3-195d-46d6-9940-e242c250d0f8)

Si tu as 5 minutes je veux bien comprendre pourquoi tous ces composants sont nécessaires.

Merci
frederic

**Comment by PricelessToolkit** [comment]
I knew this day would come, so I prepared a simulation. unzip Trigger.zip and open it in "circuitjs"

Trigger starts when the read switch state is changed to "Close to open or open to close"
1. because the "RESET Pin 2" state can be high or low C1 is for stopping current flow after "C1" is charged.
2. R8 is for limiting current on RESET Pin 2.
3. D4 is for "not doubling batt voltage every time " if it makes sense - "RESET Pin 'HIGH STATE' voltage + charged capacitor voltage".
4. R10 is for charging C1 "C1 charging slowly so that the MCU has enough time to Boot"


[Trigger.zip](https://github.com/PricelessToolkit/MailBoxGuard/files/13639264/Trigger.zip)

**Comment by fguiet** [comment]
Hi @PricelessToolkit 

Thank you for answering so quickly and providing a nice example.

Not sure to understand why the battery voltage is "doubled" without the diode...and not sure to understand quite well how the non polarized capacitor is working.

I have to search on the internet to understand better. Again thank you as with your example I am able to search and experiment by myself.

Moreover, I didnt know about circuitjs before and it's great to test an electronic circuit.

Last word, thank for sharing your knowledge because I know other guys that would not reveal their schematic and provide explanations about their projects as soon as they sell their products.

Nice piece of work !
fred

**Comment by PricelessToolkit** [comment]
Yes if the Reed switch pulls the MOSFET gate low, it will work, but it will stay ON until the door stays open. all additional components are is for turning Power ON just briefly (even if door stays open)

**Comment by fguiet** [comment]
Hi @PricelessToolkit ,

I have understood quite well how your one shot trigger is working.

It will stay ON for a bit still the capacitor is fully charged then the MOSFET gate will be HIGH and the circuit will be turn off.

The only thing I don't understand for the moment is why the voltage is doubled when the capacitor is fully charged and why the diode prevents this from happening...

Thank you again for answering me,
fred

**Reply by PricelessToolkit** [reply]
I'm not good at explaining, but I'll try. 

think like you have 2 cell battery 1 cell is ( reset pin "battery voltage") second is (Capacitor "battery voltage").
when the reset pin goes low, the capacitor starts charging through a resistor (the capacitor "-" side is the reset pin) when the reset pin changes from low to high, it becomes like 2 cell battery.  So when voltage is higher than battery voltage it has only one way to go to lower potential through diode "it discharges through diode".

**Reply by fguiet** [reply]
Ok ok, RESET PIN and Capacitor act like 2 cells in series so 3.3v + 3.3v = 6.6v. I got this. But why the diode is the only way? Maybe it is a dumb question but why it is not going through the resistance ?
By the way, I didn't thing that a diode "consumes" lot of current and could be use to discharge something.

Truly,
fred

**Reply by fguiet** [reply]
Ok I understood...if I remove the diode...it will go througth the resistance but it will take time to discharge (like it takes time to charge).....pffff sorry but I understand quickly but you have to explain me for a very long time :).

Hope this discussion will help others like me

Thank you for sharing your work!
fred

**Reply by PricelessToolkit** [reply]
I'm glad everything is clear now 👍
